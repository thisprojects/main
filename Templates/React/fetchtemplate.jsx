import React from 'react';
import Fetch from 'react-fetch';






class Welcome extends React.Component {
    render (){
        return (<h1>Hello1 {this.props.value}</h1>);


    }

}

class List extends React.Component {

                render () {
   return (
       <ul>
           {this.props.list.map(station => <li key={station}> {station} </li>)}
       </ul>
   );
}



}


class ShoppingList extends React.Component {

constructor (props){
            super(props);
            this.state = {value: "hello", schnups: false, list: ["one","two","three"], results: [] }

        }

componentDidMount () {

 this._fetchIT ();

}


  render() {



    return (
      <div className="shopping-list">
        <h1>List of Stuff</h1>

        <ul>
        <List list={this.state.list}/>
        </ul>
        <Welcome value={this.state.value}/>
        {this._doStuff()}
        {this._jsonList()}
        <div>

        </div>
        <button onClick={this._testSt.bind(this, this.state.schnups)}>Click me!</button>
      </div>

    );
  }

_doStuff () {

    if (this.state.schnups){

    return (<h1> Bibobs </h1>);
     }
}

_testSt () {
        this.setState({value: "wibbums", schnups: !this.state.schnups, list: ["one", "two", "three", "four", 13] });
        this.setState({results: []});
        this._fetchIT();

    }
_jsonList (){

    if (this.state.results.length){
        let starwars = this.state.results
        return (
            <div>

                {starwars.map (starwars  =>
                <ul>
                <li key={starwars.toString()}>{starwars.name}</li>
                </ul>
                )}
            </div>
        )


    }else{

        return (

            <div>
            <h1>Loading!...</h1>
            </div>
        )
    }


}

_fetchIT () {
    fetch('https://swapi.co/api/people/')
   .then(response => response.json())
      .then(data => this.setState({ results: data.results }));


}

}

export default ShoppingList;
