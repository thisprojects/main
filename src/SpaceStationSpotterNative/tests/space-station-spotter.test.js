import React from 'react';
import SpaceStationSpotter from '../SpaceStationSpotterNative.js';
import renderer from 'react-test-renderer';

jest.mock('react-native-maps', () => {
  const { View } = require( 'react-native' );
  const MockMapView = ( props: any ) => {
    return <View>{ props.children }</View>;
  };
  const MockMarker = ( props: any ) => {
    return <View>{ props.children }</View>;
  };
  return {
    __esModule: true,
    default: MockMapView,
    Marker: MockMarker,
  };
});

test('SpaceStationSpotter Matches Snapshot', () => {
  const component = renderer.create(
    <SpaceStationSpotter />,
  );
  let tree = component.toJSON();
  expect( tree ).toMatchSnapshot();
});
