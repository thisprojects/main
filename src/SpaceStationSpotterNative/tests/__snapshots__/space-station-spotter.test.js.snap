// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpaceStationSpotter Matches Snapshot 1`] = `
<View>
  <View>
    <View />
  </View>
  <View />
  <View
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(1, 1, 1, 0.95)",
          "display": "flex",
          "height": 1334,
          "paddingTop": 120,
          "position": "absolute",
          "top": 0,
          "width": 750,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "lightgray",
            "fontSize": 30,
            "marginBottom": 20,
          }
        }
      >
        Space Station Spotter
      </Text>
      <Text />
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(255, 255, 255, 0.9)",
              "borderRadius": 10,
              "color": "gray",
              "margin": 20,
              "opacity": 1,
              "paddingBottom": 15,
              "paddingLeft": 60,
              "paddingRight": 60,
              "paddingTop": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "gray",
                "fontSize": 17,
                "padding": 10,
              }
            }
          >
            Start Spotting!
          </Text>
        </View>
      </View>
      <View />
      <View
        style={
          Object {
            "display": "none",
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="white"
          hidesWhenStopped={true}
          size="large"
          style={
            Object {
              "padding": 20,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
